src/diagnostics/land_compute_methods.jl

```diff
@@ -0,0 +1,146 @@
+@land_compute "canopy_latent_heat_flux" SoilCanopyModel p.canopy.energy.lhf
+@land_compute "canopy_sensible_heat_flux" SoilCanopyModel p.canopy.energy.shf
+
+# Canopy - Hydraulics
+@land_compute "area_index" SoilCanopyModel p.canopy.hydraulics.area_index.leaf
+@land_compute "canopy_water_potential" SoilCanopyModel p.canopy.hydraulics.ψ
+@land_compute "cross_section" SoilCanopyModel p.canopy.hydraulics
+@land_compute "cross_section_roots" SoilCanopyModel p.canopy.hydraulics.fa_roots
+@land_compute "moisture_stress_factor" SoilCanopyModel p.canopy.hydraulics.β
+
+# Canopy - Photosynthesis
+@land_compute "photosynthesis_net_canopy" SoilCanopyModel p.canopy.photosynthesis.GPP
+@land_compute "photosynthesis_net_leaf" SoilCanopyModel p.canopy.photosynthesis.An
+@land_compute "respiration_leaf" SoilCanopyModel p.canopy.photosynthesis.Rd
+@land_compute "vcmax25" SoilCanopyModel p.canopy.photosynthesis.vcmax25
+
+# Canopy - Radiative Transfer
+@land_compute "near_infrared_radiation" SoilCanopyModel p.canopy.radiative_transfer.nir
+@land_compute "near_infrared_radiation_absorbed" SoilCanopyModel p.canopy.radiative_transfer.anir
+@land_compute "near_infrared_radiation_reflected" SoilCanopyModel p.canopy.radiative_transfer.rnir
+@land_compute "near_infrared_radiation_transmitted" SoilCanopyModel p.canopy.radiative_transfer.tnir
+@land_compute "photosynthetically_active_radiation" SoilCanopyModel p.canopy.radiative_transfer.par
+@land_compute "photosynthetically_active_radiation_absorbed" SoilCanopyModel p.canopy.radiative_transfer.apar
+@land_compute "photosynthetically_active_radiation_reflected" SoilCanopyModel p.canopy.radiative_transfer.rpar
+@land_compute "photosynthetically_active_radiation_transmitted" SoilCanopyModel p.canopy.radiative_transfer.tpar
+@land_compute "radiation_longwave_net" SoilCanopyModel p.canopy.radiative_transfer.LW_n
+@land_compute "radiation_shortwave_net" SoilCanopyModel p.canopy.radiative_transfer.SW_n
+
+# Canopy - Transpiration
+@land_compute "canopy_transpiration" SoilCanopyModel p.canopy.transpiration
+
+## Drivers Module ##
+
+@land_compute "air_co2" SoilCanopyModel p.drivers.c_co2
+@land_compute "pressure" SoilCanopyModel p.drivers.P
+@land_compute "rainfall" SoilCanopyModel p.drivers.P_liq
+@land_compute "radiation_longwave_down" SoilCanopyModel p.drivers.LW_d
+@land_compute "radiation_shortwave_down" SoilCanopyModel p.drivers.SW_d
+@land_compute "snowfall" SoilCanopyModel p.drivers.P_snow
+@land_compute "solar_zenith_angle" SoilCanopyModel p.drivers.θs
+@land_compute "specific_humidity" SoilCanopyModel p.drivers.q
+@land_compute "wind_speed" SoilCanopyModel p.drivers.u
+
+## Soil Module ##
+
+@land_compute "infiltration" SoilCanopyModel p.soil.infiltration
+@land_compute "soil_hydraulic_conductivity" SoilCanopyModel p.soil.K
+@land_compute "soil_thermal_conductivity" SoilCanopyModel p.soil.κ
+@land_compute "soil_water_liquid" SoilCanopyModel top_center_to_surface(
```

Comment:
just a note that this only returns the top surface value (2D), and not the whole field (3D), while all others return the whole field at each layer. We may want the 3D field as long as there is an easy way to plot just the surface value if we want it.

I had added this in because the diagnostics previously were making plots of 2d fields only

Reply:
I will remove it from diagnostics for now and only keep the 3D one!